nvflare/app_common/decomposers/common_decomposers.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Decomposers for types from app_common and Machine Learning libraries."""
+from abc import ABC
+from io import BytesIO
+from typing import Any
+
+import numpy as np
+import tensorflow as tf
+import torch
+from tensorflow.python.framework import ops
+
+from nvflare.app_common.abstract.learnable import Learnable
+from nvflare.fuel.utils import fobs
+
+
+class LearnableDecomposer(fobs.Decomposer):
+    @staticmethod
+    def supported_type():
+        return Learnable
+
+    def decompose(self, target: Learnable) -> Any:
+        return target.copy()
+
+    def recompose(self, data: Any) -> Learnable:
+        obj = Learnable()
+        for k, v in data.items():
+            obj[k] = v
+        return obj
+
+
+class NumpyTypeDecomposer(fobs.Decomposer, ABC):
+    """Decomposer base class for all numpy types with item method."""
+
+    def decompose(self, target: Any) -> Any:
+        return target.item()
+
+    def recompose(self, data: Any) -> np.ndarray:
+        return self.supported_type()(data)
+
+
+class Float64TypeDecomposer(NumpyTypeDecomposer):
+    @staticmethod
+    def supported_type():
+        return np.float64
+
+
+class Float32TypeDecomposer(NumpyTypeDecomposer):
+    @staticmethod
+    def supported_type():
+        return np.float32
+
+
+class Int64TypeDecomposer(NumpyTypeDecomposer):
+    @staticmethod
+    def supported_type():
+        return np.int64
+
+
+class Int32TypeDecomposer(NumpyTypeDecomposer):
+    @staticmethod
+    def supported_type():
+        return np.int32
+
+
+class ArrayDecomposer(fobs.Decomposer):
```

Comment:
Suggest rename to `NumpyArrayDecomposer`

Reply:
Good idea.

Reply:
Good idea.